button/internal/button.ts

```diff
@@ -67,6 +67,13 @@ export abstract class Button extends buttonBaseClass implements FormSubmitter {
    */
   @property() href = '';
 
+  /**
+   * The filename to use when downloading the linked resource. 
+   * If not specified, the browser will determine a filename. 
+   * This is only applicable when the button is used as a link (`href` is set).
+   */
+  @property() downloadFilename = '';
```

Comment:
Can you rename this to `download`, to match the [HTMLAnchorElement.download](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement/download) property?

It would also be super helpful to include and forward this property for [link chips](https://github.com/material-components/material-web/blob/ac916578a26ea8b6b19ad2e9fdeb3f4970c2e993/chips/internal/assist-chip.ts#L51) and [link icon buttons](https://github.com/material-components/material-web/blob/ac916578a26ea8b6b19ad2e9fdeb3f4970c2e993/iconbutton/internal/icon-button.ts#L193).